docs/proposals/device-management-enhance.md

```diff
@@ -0,0 +1,1069 @@
+---
+title: Device Management Enhancement
+authors:
+    - "@luogangyi"
+approvers:
+    - "@kevin-wangzefeng"
+    - "@fisherxu"
+creation-date: 2020-05-13
+last-updated: 2020-07-16
+status: implementable
+---
+
+# Device Management Enhancement
+## Motivation
+
+Device management is a key feature required for IoT use-cases in edge computing.
+This proposal addresses how can we enhance device management including more flexible
+protocol support, more reasonable device model and device instance design, and
+additional way to handle data which is collected from device.
+
+### Goals
+
+* Add customized protocol support
+* Add ability to allow user get and process data in edge node
+* Improve device model and device instance crd design
+
+### Non-goals
+
+* To support streaming data from video device
+* To provide specific protocols and mappers
+
+## Proposal
+
+We propose below modifications on current design.
+* Move property visitors from device model to device instance.
+* Add collectCycle and reportCycle under property visitor.
+* Add data section in twin spec.
+* Add customized protocol config in device model and device instance CRDs.
+* Add common part in protocol config section
+* Extract common part of Modbus protocol config into an independent common part.
+* Allow add any customized K-V in protocol config and property visitor section.
+
+
+
+### Use Cases
+
+* Reuse device model.
+  * Considering device properties are physical attributes, but property visitors are manually configured
+attributes. Combining device properties and property visitors in device model reduce the reusability of device model.
+  * Case 1: Same devices are connected to a central management server, eg. SCADA. In this case, devices have same properties but
```

Comment:
Just a small question :)  Same devices are connected to a central management server, they will also have different property visitors?